site/src/main/java/uk/nhs/digital/website/beans/Role.java

```diff
@@ -17,21 +15,34 @@ public String getSectionType() {
         return "role";
     }
 
+    @HippoEssentialsGenerated(internalName = "website:primaryrolepicker")
+    public HippoBean getPrimaryrolepicker() {
+        return getLinkedBean("website:primaryrolepicker", HippoBean.class);
+    }
+
     @HippoEssentialsGenerated(internalName = "website:primaryrole")
     public String getPrimaryrole() {
+        JobRole primaryRolePicker = (JobRole)this.getPrimaryrolepicker();
+        if (primaryRolePicker != null) {
+            return primaryRolePicker.getTitle();
+        }
         return getProperty("website:primaryrole");
     }
 
+    @HippoEssentialsGenerated(internalName = "website:primaryroleorgpicker")
+    public HippoBean getPrimaryroleorgpicker() {
+        return getLinkedBean("website:primaryroleorgpicker", HippoBean.class);
+    }
+
     @HippoEssentialsGenerated(internalName = "website:primaryroleorg")
```

Comment:
Does this method deviate from the auto generated bean? If it does I normally add allowModifications = false to the @HippoEssentialsGenerated annotation. I think doing so should stop your modification being lost.